snow/engine/avalanche/vertex/builder_test.go

```diff
@@ -0,0 +1,50 @@
+package vertex
+
+import (
+	"testing"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBuildInvalid(t *testing.T) {
+	chainID := ids.ID{1}
+	height := uint64(2)
+	epoch := uint32(0)
+	parentIDs := []ids.ID{{4}, {5}}
+	txs := [][]byte{{7}, {6}}
+	restrictions := []ids.ID{{8}, {9}}
+	_, err := Build(
+		chainID,
+		height,
+		epoch,
+		parentIDs,
+		txs,
+		restrictions,
+	)
+	assert.Error(t, err, "Build should have errored, but didn't")
```

Comment:
Could you add in the reason this test should have errored?

Reply:
Is this due to their being restrictions in epoch 0?

Reply:
added